source/tyr/tyr/resources.py

```diff
@@ -272,8 +272,10 @@ def get(self, id=None, name=None):
         parser.add_argument('is_free', type=inputs.boolean, required=False,
                 case_sensitive=False, help='boolean for returning only free or private instances')
         args = parser.parse_args()
-        if args['is_free'] != None:
-            return models.Instance.query.filter_by(**args).all()
+        args.update({'id': id, 'name': name})
+        if any(v is not None for v in args.values()):
+            filters = {k: v for k, v in args.items() if v is not None}
```

Comment:
Sorry for bothering you again, do you really need this line? Will it still working if you remove this line?